pkg/apis/controller/experiments/v1beta1/experiment_types.go

```diff
@@ -187,8 +188,36 @@ type FeasibleSpace struct {
 }
 
 type TrialTemplate struct {
-	Retain     bool        `json:"retain,omitempty"`
+	// Retain indicates that Trial resources must be not cleanup
+	Retain bool `json:"retain,omitempty"`
+
 	GoTemplate *GoTemplate `json:"goTemplate,omitempty"`
```

Comment:
Can we move it into TrialSpec?

Reply:
In that case what do you think about the name where `Unstructured` template will be located?
```
type TrialSpec struct {
<must be some name>     *unstructured.Unstructured `json:"<name>,omitempty"`

ConfigMapName string `json:"configMapName,omitempty"`
ConfigMapNamespace string `json:"configMapNamespace,omitempty"`
TemplatePath string `json:"templatePath,omitempty"`
}

Reply:
How about this:

```go
type TrialTemplate struct {
	Retain     bool        `json:"retain,omitempty"`

	TrialSource `json:",inline"`

	// List of parameres that are used in Trial template
	TrialParameters []TrialParameterSpec `json:"trialParameters,omitempty"`
}

type TrialSource struct {
	ConfigMap *ConfigMapSource `json:"configMap"`
	TrialSpec *unstructured.Unstructured `json:"trialSPec"`
}

type ConfigMapSource struct {
	// Name of config map where Trial template is located
	ConfigMapName string `json:"configMapName,omitempty"`

	// Namespace of config map where Trial template is located
	ConfigMapNamespace string `json:"configMapNamespace,omitempty"`
	
	// Path in config map where Trial template is located
	TemplatePath string `json:"templatePath,omitempty"`
}
```

Reply:
@gaocegege SGTM, it is nice to follow k8s APIs structure way.
What do you think @johnugeorge ?

Reply:
Looks great